### PR TITLE
feat: log error cause, inpro-1831

### PR DESCRIPTION
**Pull request by andrewYevtushenk**
# Jira

Jira ticket - [INPRO-1831](https://strg-office.atlassian.net/browse/INPRO-1831)

## Description

Support for recursive logging error cause with stacktraces

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation
- [ ] Refactoring
- [ ] Performance
- [ ] Hotfix

## Tests

- [ ] Unit tests have been added
- [ ] Other kinds of tests have been added
- [x] The changes have been tested manually by `@SLACK_USERNAME`

[INPRO-1831]: https://strg-office.atlassian.net/browse/INPRO-1831?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -17,7 +17,7 @@
       "allowUnusedLabels": false,
       "declaration": true,
       "forceConsistentCasingInFileNames": true,
-      "lib": ["es2018"],
+      "lib": ["es2022"],
```
Comment (andrewYevtushenk) [inline tsconfig.json:20]: In ES2018 `cause` property does not exist in Error

**Comment by andrewYevtushenk** [comment]
Example of `simpleFormat`

Screenshot 2024-02-07 at 14 46 23


Example of `jsonFormat`

`{"level":"error","message":"Controller Error","metadata":{},"stack_trace":{"cause":{"cause":{"message":"Dao Error","stack":"Error: Dao Error\n    at Object.<anonymous> (/Users/andriiyevtuchenko/Desktop/STRG/logging-winston/build/test.js:4:90)\n    at Module._compile (node:internal/modules/cjs/loader:1376:14)\n    at Module._extensions..js (node:internal/modules/cjs/loader:1435:10)\n    at Module.load (node:internal/modules/cjs/loader:1207:32)\n    at Module._load (node:internal/modules/cjs/loader:1023:12)\n    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:135:12)\n    at node:internal/main/run_main_module:28:49"},"message":"Service Error","stack":"Error: Service Error\n    at Object.<anonymous> (/Users/andriiyevtuchenko/Desktop/STRG/logging-winston/build/test.js:4:54)\n    at Module._compile (node:internal/modules/cjs/loader:1376:14)\n    at Module._extensions..js (node:internal/modules/cjs/loader:1435:10)\n    at Module.load (node:internal/modules/cjs/loader:1207:32)\n    at Module._load (node:internal/modules/cjs/loader:1023:12)\n    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:135:12)\n    at node:internal/main/run_main_module:28:49"},"message":"Controller Error","stack":"Error: Controller Error\n    at Object.<anonymous> (/Users/andriiyevtuchenko/Desktop/STRG/logging-winston/build/test.js:4:15)\n    at Module._compile (node:internal/modules/cjs/loader:1376:14)\n    at Module._extensions..js (node:internal/modules/cjs/loader:1435:10)\n    at Module.load (node:internal/modules/cjs/loader:1207:32)\n    at Module._load (node:internal/modules/cjs/loader:1023:12)\n    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:135:12)\n    at node:internal/main/run_main_module:28:49"}}`

File: src/index.ts [hunk 2]
```diff
@@ -15,31 +20,60 @@ winston.addColors({
   debug: 'grey',
 })
 
-const displayStackSimple = winston.format((info: TransformableInfo) => {
-  info.message = info.stack ? info.stack : info.message
-  return info
+const displayStackSimple = winston.format((info: any) => {
+  if (info.level !== 'error') {
```
Comment (EvolvingApe) [inline src/index.ts:24]: can there be levels > error like "critical" ?

Comment (hhofstaetter) [reply src/index.ts:24]: https://github.com/winstonjs/winston?tab=readme-ov-file#logging-levels

seems like winston uses error as the highest level (as default)

**Review by hhofstaetter — APPROVED** [review]

**Review by EvolvingApe — APPROVED** [review]

**Pull request merged**